pegasus/helpers/section_api_helpers.rb

```diff
@@ -254,9 +303,18 @@ def self.assignable_info(course_or_script, hidden=false)
   # @param user_id [Integer]
   # @return AssignableInfo[]
   def self.valid_scripts(user_id = nil)
+    has_any_experiment = DashboardCourseExperiments.has_any_experiment?(user_id)
+    # Users with course experiments enabled effectively lose their hidden
+    # script access permissions to avoid unnecessary complexity.
+    with_hidden = !has_any_experiment && user_id && Dashboard.hidden_script_access?(user_id)
+    scripts = valid_default_scripts(user_id, with_hidden)
+    return scripts unless has_any_experiment
+    scripts.each {|script| set_alternate_script_info(user_id, script)}
+  end
```

Comment:
It may be too late to go down this road, but I wonder if this would be simpler if we sent all clients both the default and experiment scripts, also sent them a list of what experiments they were in, and then let the client do the filtering.

Reply:
The short answer is I did consider this briefly, and perhaps I should have raised this as a question / with a mini design doc before proceeding, but I went with this implementation and without fully exploring each option primarily due to the deadline.

Moving forward, it may be less long-term badness to do the script-swapping on the client than what I have here, but it's worth considering whether we want to live with that client-side logic indefinitely or whether we ultimately want the logic to end up in `/dashboardapi/valid_scripts` in dashboard server. I would argue that dashboard server is the best final destination for this code, and I think it's quite doable to move it there, I just didn't want to risk it on a short deadline.

I am of the opinion that it is worth fully validating the feasibility of moving this API to dashboard right after we hit the subgoals deadline requirements, while it is still fresh in our minds.